src/adapters/Cornerstone3D/Bidirectional.js

```diff
@@ -0,0 +1,203 @@
+
+        [longAxisSCOORDGroup, shortAxisSCOORDGroup].forEach(group => {
+            const { GraphicData } = group;
+            for (let i = 0; i < GraphicData.length; i += 2) {
+                const point = imageToWorldCoords(imageId, [
+                    GraphicData[i],
+                    GraphicData[i + 1]
+                ]);
+                worldCoords.push(point);
+            }
+        });
+
+        const state = {
+            ...defaultState,
+            finding: findingGroup
+                ? findingGroup.ConceptCodeSequence
+                : undefined,
+            findingSites: findingSiteGroups.map(fsg => {
+                return { ...fsg.ConceptCodeSequence };
+            }),
+            toolType: Bidirectional.toolType,
+            data: {
+                handles: {
+                    points: [
+                        worldCoords[0],
+                        worldCoords[1],
+                        worldCoords[2],
+                        worldCoords[3]
+                    ],
+                    activeHandleIndex: 0,
+                    textBox: {
+                        hasMoved: false
+                    }
+                },
+                cachedStats: {
+                    [`imageId:${imageId}`]: {
+                        length:
+                            longAxisNUMGroup.MeasuredValueSequence.NumericValue,
+                        width:
+                            shortAxisNUMGroup.MeasuredValueSequence.NumericValue
+                    }
+                }
+            }
+        };
+
+        return state;
+    }
+
+    static getTID300RepresentationArguments(tool, worldToImageCoords) {
+        const { data, finding, findingSites, metadata } = tool;
+        const { cachedStats, handles } = data;
+
+        const { referencedImageId } = metadata;
+
+        if (!referencedImageId) {
+            throw new Error(
+                "Bidirectional.getTID300RepresentationArguments: referencedImageId is not defined"
+            );
+        }
+
+        const { length, width } = cachedStats[`imageId:${referencedImageId}`];
+        const { points } = handles;
+
+        // Find the length and width point pairs by comparing the distances of the points at 0,1 to points at 2,3
+        let firstPointPairs = [points[0], points[1]];
+        let secondPointPairs = [points[2], points[3]];
+
+        let firstPointPairsDistance = Math.sqrt(
+            Math.pow(firstPointPairs[0][0] - firstPointPairs[1][0], 2) +
+                Math.pow(firstPointPairs[0][1] - firstPointPairs[1][1], 2) +
+                Math.pow(firstPointPairs[0][2] - firstPointPairs[1][2], 2)
+        );
+
+        let secondPointPairsDistance = Math.sqrt(
+            Math.pow(secondPointPairs[0][0] - secondPointPairs[1][0], 2) +
+                Math.pow(secondPointPairs[0][1] - secondPointPairs[1][1], 2) +
+                Math.pow(secondPointPairs[0][2] - secondPointPairs[1][2], 2)
+        );
+
+        let lengthPointPairs;
```

Comment:
I know these were called this before in the old implementation, but can we call these `longAxis` and `shortAxis` instead? I think its semantically better and easier to follow that they become the long and short axes.

Reply:
sure